app/api/public-cloud/aws-roles/helpers.ts

```diff
@@ -40,6 +46,26 @@ const paginate = <T>(users: T[], options: PaginationOptions): T[] => {
   return users.slice(startIndex, endIndex);
 };
 
+const isUpperCase = (char: string): boolean => {
+  const charCode = char.charCodeAt(0);
+  return charCode >= 65 && charCode <= 90;
+};
+
+const parseGroupNameToTab = (name: string): tabName => {
```

Comment:
Wonder if what the `group name` looks like?
In my view, the process to generate & parse the role names are not intuitive, we might want to revisit the architectural decision to make it easier to understand.

Reply:
Group name will be like "BillingViewrs" or "Admins" and they can't rename it

Reply:
add comments to explain